server/src/test/java/org/elasticsearch/common/util/set/SetsTests.java

```diff
@@ -94,6 +95,11 @@ public void testCapacityIsEnoughForSetToNotBeResized() {
         }
     }
 
+    public void testAddToCopy() {
+        assertThat(addToCopy(Set.of("a", "b"), "c"), containsInAnyOrder("a", "b", "c"));
```

Comment:
We use randomized testing, so I guess it would be nice to use it here, too, for consistency sake. What do you think about
something like this?
```java
        Set<String> set = randomSet(1, 100, () -> randomAlphaOfLength(16));
        String newElement = randomAlphaOfLength(10);
        assertEquals(addToCopy(set, newElement), Sets.union(set, Set.of(newElement)));
```